ondewo_bpi_qa/bpi_qa_server.py

```diff
@@ -160,12 +275,85 @@ async def send_to_cai(self, request: DetectIntentRequest,) -> Tuple[DetectIntent
             None, self.client.services.sessions.detect_intent, request,
         )
         intent_name_cai = cai_response.query_result.intent.display_name
-        logger_console.warning(
+        logger_console.debug(
             {
                 "message": f"CAI-DetectIntentResponse from CAI, intent_name_cai: {intent_name_cai}",
                 "content": intent_name_cai,
                 "intent_name_cai": intent_name_cai,
                 "tags": ["text"],
             }
         )
-        return cai_response, "cai_response"
+        return cai_response, CAI_RESPONSE_NAME
+
+    def create_session_if_not_exists(self, request: DetectIntentRequest, ) -> None:
+
+        try:
+            self.client.services.sessions.get_session(
+                request=GetSessionRequest(session_id=request.session, session_view=Session.View.VIEW_SPARSE)
```

Comment:
Is this a good thing that when you get a session, it throws an error if there is none?
I mean I know it's the existing system behaviour but I am not a fan of that

Reply:
I think the original flaw is the fact that Sessions are not created explicitly.
If we could guarantee that you cannot use `DetectIntent` without a created session, this logic wouldn't need to exist.